src/network/networkd-ndisc.c

```diff
@@ -199,7 +199,12 @@ static int ndisc_request_route(Route *in, Link *link, sd_ndisc_router *rt) {
                 r = sd_ndisc_router_get_hop_limit(rt, &hop_limit);
                 if (r < 0 && r != -ENODATA)
                         return log_link_warning_errno(link, r, "Failed to get default router hop limit from RA: %m");
-        }
+
+                link->network->ipv6_hop_limit = hop_limit;
+                r = link_set_ipv6_hop_limit(link);
+                if (r < 0)
+                        log_link_warning_errno(link, r, "Cannot set IPv6 hop limit for interface, ignoring: %m");
```

Comment:
I don't think log messages like
```
host0: Cannot set IPv6 hop limit for interface, ignoring: Invalid argument
```
are as useful as they can be. networkd should probably log the hop limit too.